applications/rag/README.md

```diff
@@ -1,192 +1,210 @@
-
-5. Replace the variables in the 1st cell with your Kaggle credentials (can be found in the `kaggle.json` file created by Step 4):
-    * `KAGGLE_USERNAME`
-    * `KAGGLE_KEY`
-
-6. Run all the cells in the notebook. This generates vector embeddings for the input dataset (`denizbilginn/google-maps-restaurant-reviews`) and stores them in the `pgvector-instance` via a Ray job.
-    * When the last cell says the job has succeeded (eg: `Job 'raysubmit_APungAw6TyB55qxk' succeeded`), the vector embeddings have been generated and we can launch the frontend chat interface.
-    * Ray may take several minutes to create the runtime environment. During this time, the job will appear to be missing (e.g. `Status message: Job has not started yet`).
-
-## Launch the frontend chat interface
-
-### With IAP disabled
-1. Setup port forwarding for the frontend: `kubectl port-forward service/rag-frontend -n $NAMESPACE 8080:8080 &`
-
-2. Go to `localhost:8080` in a browser & start chatting! This will fetch context related to your prompt from the vector embeddings in the `pgvector-instance`, augment the original prompt with the context & query the inference model (`mistral-7b`) with the augmented prompt.
-
-### With IAP enabled
-1. Verify that IAP is enabled on [Google Cloud Platform (GCP) IAP](https://console.cloud.google.com/security/iap)(make sure you are logged in) for your application. If you encounter any errors, try re-enabling IAP.
-2. From *Google Cloud Platform IAP*, check if the target user has role `IAP-secured Web App User`. This role is necessary to access the application through IAP.
-3. Verify the domain is active using command:
-    `kubectl get managedcertificates frontend-managed-cert -n rag --output jsonpath='{.status.domainStatus[0].status}'`
-4. Read terraform output: `terraform output frontend_uri` to find the domain created by IAP for accessing your service.
-5. Open your browser and navigate to the domain you retrieved in the previous step to start chatting!
-
-## Prompt examples
-
-*TODO:* Add some example prompts for the dataset.
-
-## Cleanup
-
-1. Run `terraform destroy --var-file="workloads.tfvars"`
+1. Troubleshoot Ray job failures: 
+    - If the Ray actors fail to be scheduled, it could be due to a stockout or quota issue.
+        - Run `kubectl get pods -n ${NAMESPACE:?}`. There should be a Ray head and Ray worker pod in `Running` state (prefixed by `ray-cluster-kuberay-head-` and `ray-cluster-kuberay-worker-workergroup-`). If your ray pods aren't running, it's likely due to quota or stockout issues. Check that your project and selected `cluster_location` have L4 GPU capacity.
```

Comment:
Instead of "There should be a Ray pod with prefix foo", let's use the label selector:

`kubectl get pods -n $NAMESPACE -l app.kubernetes.io/name=kuberay`